src/main/java/org/jenkinsci/plugins/workflow/cps/DSL.java

```diff
@@ -179,6 +193,26 @@ protected Object invokeStep(StepDescriptor d, Object args) {
             d.checkContextAvailability(context);
             Thread.currentThread().setContextClassLoader(CpsVmExecutorService.ORIGINAL_CONTEXT_CLASS_LOADER.get());
             s = d.newInstance(ps.namedArgs);
+            try {
+                // No point storing empty arguments, and ParallelStep is a special case where we can't store its closure arguments
+                if (ps.namedArgs != null && !(ps.namedArgs.isEmpty()) && isKeepStepInfo() && !(s instanceof ParallelStep)) {
+                    Map<String, Object> actualArgs = ps.namedArgs;
+                    if (d.isMetaStep() && ps.namedArgs.get("delegate") instanceof Map) {
+                        actualArgs = (Map<String,Object>)ps.namedArgs.get("delegate");
```

Comment:
@jglick One can simply look at the FlowNode/descriptor and see that we have unwrapped a MetaStep.  Failing to unwrap it makes it more work to consume this API because it forces the API consumer to do these inspections to figure out how to get what we actually care about (the arguments given). 

As it stands, we only have CoreStep and CoreWrapperStep that are making use of the metastep approach, and they share the delegate parameter.  AFAIK we do not have any other planned implementations using this besides testcases.   

Based on the above it would be extremely foolish to add new-new APIs to support a use case that does not exist and AFAIK will never exist. 

Instead what I can do is harden this slightly -- we can use existing StepDescriptor methods to confirm we only have a single argument, and then unwrap that (otherwise persisting everything intact).  

In the purely *hypothetical* case that we add another metastep, which has a different delegate parameter name *and* takes another argument then it will be handled seamlessly.